src/pytorch_lightning/plugins/io/torch_plugin.py

```diff
@@ -11,86 +11,23 @@
-        """Save model/training states as a checkpoint file through state-dump and file-write.
-
-        Args:
-            checkpoint: dict containing model and trainer state
-            path: write-target path
-            storage_options: not used in ``TorchCheckpointIO.save_checkpoint``
-
-        Raises:
-            TypeError:
-                If ``storage_options`` arg is passed in
-        """
-        if storage_options is not None:
-            raise TypeError(
-                "`Trainer.save_checkpoint(..., storage_options=...)` with `storage_options` arg"
-                f" is not supported for `{self.__class__.__name__}`. Please implement your custom `CheckpointIO`"
-                " to define how you'd like to use `storage_options`."
-            )
-        fs = get_filesystem(path)
-        fs.makedirs(os.path.dirname(path), exist_ok=True)
-        try:
-            # write the checkpoint dictionary on the file
-            atomic_save(checkpoint, path)
-        except AttributeError as err:
-            # todo (sean): is this try catch necessary still?
-            # https://github.com/Lightning-AI/lightning/pull/431
-            key = pl.LightningModule.CHECKPOINT_HYPER_PARAMS_KEY
-            checkpoint.pop(key, None)
-            rank_zero_warn(f"Warning, `{key}` dropped from checkpoint. An attribute is not picklable: {err}")
-            atomic_save(checkpoint, path)
-
-    def load_checkpoint(
-        self, path: _PATH, map_location: Optional[Callable] = lambda storage, loc: storage
-    ) -> Dict[str, Any]:
-        """Loads checkpoint using :func:`torch.load`, with additional handling for ``fsspec`` remote loading of
-        files.
-
-        Args:
-            path: Path to checkpoint
-            map_location: a function, :class:`torch.device`, string or a dict specifying how to remap storage
-                locations.
-
-        Returns: The loaded checkpoint.
-
-        Raises:
-            FileNotFoundError: If ``path`` is not found by the ``fsspec`` filesystem
-        """
-
-        # Try to read the checkpoint at `path`. If not exist, do not restore checkpoint.
-        fs = get_filesystem(path)
-        if not fs.exists(path):
-            raise FileNotFoundError(f"Checkpoint at {path} not found. Aborting training.")
-
-        return pl_load(path, map_location=map_location)
+    respectively, common for most use cases.
 
-    def remove_checkpoint(self, path: _PATH) -> None:
-        """Remove checkpoint file from the filesystem.
+    .. deprecated:: v1.8.0     This plugin has been deprecated in v1.8.0 and will be removed in v1.10.0.
+    """
 
-        Args:
-            path: Path to checkpoint
-        """
-        fs = get_filesystem(path)
-        if fs.exists(path):
-            fs.rm(path, recursive=True)
-            log.debug(f"Removed checkpoint: {path}")
+    def __new__(cls, *args: Any, **kwargs: Any) -> "TorchCheckpointIO":
+        rank_zero_deprecation(
```

Comment:
@Borda @awaelchli

I thought we agreed we won’t add any warning for all classes moved to `lightning_lite`?

As a user, I don’t mind if the TorchCheckpointIO is coming from lightning_lite. I still want to be able to import it from PyTorch Lightning without warning.

Reply:
For now, I removed the deprecations. We can add them back later if we feel it is a good time.